src/AzureClient/AzureClient.cs

```diff
@@ -572,10 +585,31 @@ public async Task<ExecutionResult> GetJobResultAsync(IChannel? channel, string j
                 //                 cancellation token support.
                 var request = WebRequest.Create(job.OutputDataUri);
                 using var responseStream = (await request.GetResponseAsync()).GetResponseStream();
-                return responseStream.ToHistogram(
-                    Logger, 
-                    isSimulatorOutput: this.ActiveTarget?.TargetId?.StartsWith(MicrosoftSimulator) ?? false)
-                .ToExecutionResult();
+                if (this.ActiveTarget?.TargetId?.StartsWith(MicrosoftSimulator) ?? false)
+                {
+                    var outputLines = ReadAllLines(responseStream).ToList();
+                    outputLines = outputLines.Select(line => line.Trim()).ToList();
+                    
+                    // ToDo: This is a hack to separate Messages from the result.
+                    var resultStartLine = outputLines.Count() - 1;
+                    if (outputLines[resultStartLine].EndsWith('"'))
+                    {
+                        while (!outputLines[resultStartLine].StartsWith('"'))
+                        {
+                            resultStartLine -= 1;
+                        }
+                    }
+                    
+                    var messages = String.Join('\n', outputLines.Take(resultStartLine));
+                    channel?.Stdout(messages);
+                    var result = String.Join(' ', outputLines.Skip(resultStartLine));
+
+                    return result.ToExecutionResult();
```

Comment:
I think it would be good to encapsulate the code that differentiates messages from result into a single method that returns a tuple:
```csharp
private static (string Messages, string Result) ParseSimulatorOutput(Stream stream)
```

This way the only thing that has to go inside the `if` statement is:
```csharp
var parsedOutput = ParseSimulatorOutput(Stream stream);
channel?.Stdout(parsedOutput.messages);
parsedOutput.Result.ToExecutionResult();
```

Reply:
done